src/controllers/selectSignatories.controller.ts

```diff
@@ -0,0 +1,126 @@
+import { BAD_REQUEST, OK } from 'http-status-codes'
+import { inject } from 'inversify'
+import { controller, httpGet, httpPost, requestBody } from 'inversify-express-utils'
+import { RedirectResult } from 'inversify-express-utils/dts/results'
+import BaseController from './base.controller'
+
+import DirectorToSignMapper from 'app/mappers/check-your-answers/directorToSign.mapper'
+import SelectSignatoriesFormModel from 'app/models/form/selectSignatories.model'
+import Optional from 'app/models/optional'
+import DirectorToSign from 'app/models/session/directorToSign.model'
+import DissolutionSession from 'app/models/session/dissolutionSession.model'
+import DirectorDetails from 'app/models/view/directorDetails.model'
+import ValidationErrors from 'app/models/view/validationErrors.model'
+import { DEFINE_SIGNATORY_INFO_URI, SELECT_SIGNATORIES_URI } from 'app/paths'
+import selectSignatoriesSchema from 'app/schemas/selectSignatories.schema'
+import CompanyOfficersService from 'app/services/company-officers/companyOfficers.service'
+import SessionService from 'app/services/session/session.service'
+import TYPES from 'app/types'
+import FormValidator from 'app/utils/formValidator.util'
+
+interface ViewModel {
+  signatories: DirectorDetails[]
+  data?: Optional<SelectSignatoriesFormModel>
+  errors?: Optional<ValidationErrors>
+}
+
+@controller(SELECT_SIGNATORIES_URI, TYPES.SessionMiddleware, TYPES.AuthMiddleware, TYPES.CompanyAuthMiddleware)
+export class SelectSignatoriesController extends BaseController {
+
+  public constructor(
+    @inject(SessionService) private session: SessionService,
+    @inject(CompanyOfficersService) private officerService: CompanyOfficersService,
+    @inject(FormValidator) private validator: FormValidator,
+    @inject(DirectorToSignMapper) private mapper: DirectorToSignMapper) {
+    super()
+  }
+
+  @httpGet('')
+  public async get(): Promise<string> {
+    const session: DissolutionSession = this.session.getDissolutionSession(this.httpContext.request)!
+
+    const signatories: DirectorDetails[] = await this.getSignatories(session.selectDirectorForm!.director!)
+
+    return this.renderView(signatories, session.selectSignatoriesForm)
+  }
+
+  @httpPost('')
+  public async post(@requestBody() body: SelectSignatoriesFormModel): Promise<string | RedirectResult> {
+    if (typeof body.signatories === 'string') {
+      body.signatories = [body.signatories]
+    }
```

Comment:
this is a small niche of checkboxes. When you submit a single checkbox, it comes through as a string. When you submit multiple checkboxes, it comes through as a string[]. This is a small bit of code to ensure that we only need to treat them as string[] to reduce complexity.